src/utils/point_locator_nanoflann.C

```diff
@@ -85,17 +85,24 @@ PointLocatorNanoflann::init ()
           _ids = std::make_shared<std::vector<dof_id_type>>();
           _point_cloud = std::make_shared<std::vector<Point>>();
 
-          // Make the KD-Tree out of mesh element centroids.
-
-          // We can either reserve exactly the right amount of space or
-          // let push_back() take care of it, not sure what would be
-          // faster actually, since it takes some time to count the number
-          // of active+local elements.
-          auto n_active_local_elem = _mesh.n_active_local_elem();
-          _ids->reserve(n_active_local_elem);
-          _point_cloud->reserve(n_active_local_elem);
-
-          for (const auto & elem : _mesh.active_local_element_ptr_range())
+          // Make the KD-Tree out of active element centroids.
+          //
+          // Note: we use active elements rather than active+local
+          // elements since we sometimes need to be able to locate
+          // points in ghosted elements (for example in Periodic BCs).
+          // Active elements are also natural to use in the
+          // DistributedMesh case. The size of the KD-Tree on each
+          // processor therefore scales with the number of elements on
+          // each processor in either case.
+          //
+          // Note: we reserve enough space for n_elem() entries rather
+          // than counting the exact number of active elements, which
+          // would require an extra element loop.
```

Comment:
Yikes - that's a *lot* of unnecessary space reserved on problems with large core counts.  I'd feel better just letting the containers' own logarithmic resizing work, if you don't want to do the extra loop, so we only reserve at most 2 times too much space instead of N_procs times too much.

Reply:
Ah, I think I got confused thinking there was a difference between `n_elem()` and `parallel_n_elem()` or something... this is a bad idea for `DistributedMesh` of course so I'll fix that.